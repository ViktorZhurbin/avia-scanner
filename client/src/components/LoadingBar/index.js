import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import styles from './index.css';

const cx = cl.bind(styles);

const LoadingBar = ({ isLoading }) => (
    <div className={cx({ isLoading })} />
);

LoadingBar.propTypes = {
    isLoading: PropTypes.bool.isRequired,
};

export default React.memo(LoadingBar);
