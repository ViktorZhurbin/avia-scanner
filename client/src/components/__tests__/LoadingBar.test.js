import React from 'react';
import renderer from 'react-test-renderer';

import LoadingBar from '../LoadingBar';

describe('Render LoadingBar', () => {
    it('render correctly', () => {
        const component = renderer.create(
            <LoadingBar isLoading />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
