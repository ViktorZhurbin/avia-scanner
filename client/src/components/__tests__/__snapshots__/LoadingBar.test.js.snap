// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LoadingBar render correctly 1`] = `
<div
  className="isLoading"
/>
`;
